pkg/bearings/magnetic.go

```diff
@@ -0,0 +1,53 @@
+package bearings
+
+import (
+	"math"
+
+	"github.com/martinlindhe/unit"
+)
+
+type Magnetic struct {
+	θ unit.Angle
+}
+
+func NewMagneticBearing(value unit.Angle) *Magnetic {
+	return &Magnetic{θ: Normalize(value)}
+}
+
+var _ Bearing = &Magnetic{}
+
+func (b *Magnetic) Value() unit.Angle {
+	return Normalize(b.θ)
+}
+
+func (b *Magnetic) Rounded() unit.Angle {
+	return unit.Angle(b.RoundedDegrees()) * unit.Degree
+}
+
+func (b *Magnetic) Degrees() float64 {
+	return b.Value().Degrees()
+}
+
+func (b *Magnetic) RoundedDegrees() float64 {
+	return math.Round(b.Degrees())
+}
+
+func (b *Magnetic) True(declination unit.Angle) Bearing {
+	return NewTrueBearing(b.Value() + declination)
+}
+
+func (b *Magnetic) Magnetic(declination unit.Angle) Bearing {
+	return b
+}
+
+func (b *Magnetic) IsTrue() bool {
+	return false
+}
+
+func (b *Magnetic) IsMagnetic() bool {
+	return true
+}

```

Comment:
This is surprising to me, are we going to be checking if a given bearing is True/Mag every time we are dealing with this type? I would have expected Bearing to handle the differences on its own.

Reply:
How would a Bearing know if it's being used in a context where we need a True heading (ACMI data, messages streamed from the `Sim` interface, calls into the `geo` package) or a Magnetic heading (brevity input/output)? We have to tell it that information somehow.

We should prefer to use `Magnetic(unit.Angle)` and `True(unit.Angle)` over `IsTrue()`/`IsMagnetic()`, in any case. There are a few contexts where we don't have access to the current time and location, and therefore don't know the declination, those are the only places we bother checking, and ideally we convert before calling those functions.

If we do know the declination, no reason to check what we started with- convert immediately to what we need.